### PR TITLE
added get, delete, put

### LINKED ISSUE #14: Implement API for getting, updating and deleting individual users
Implement API for getting, updating and deleting individual users as following:
DELETE users/[id]
PUT users/[id]
GET users/[id]

Test it on users with id > 3

### PR DESCRIPTION
**Pull request by andy-sam**
Fixes #14

### CODE REVIEW

**Comment by andy-sam** [comment]
I have fix put Request according to your comments.
It works like this: the record updated if Id from the body request already exists and added a new record in another case.

File: src/main/java/com/codesarge/banking/controllers/UserController.java [hunk 1]
```diff
@@ -28,9 +28,32 @@ public UserController(UserService userService) {
...
+        return userService.findUserById( id );
+    }
+
     @PostMapping("users")
     public UserDto addUser(@RequestBody User user) {
         log.info("Requested creating user: " + user);
         return userService.addUser(user);
     }
+
+    @PutMapping("/users/{id}")
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/controllers/UserController.java:43]: let's get rid of id in this mapping

File: src/main/java/com/codesarge/banking/controllers/UserController.java [hunk 2]
```diff
@@ -28,9 +28,32 @@ public UserController(UserService userService) {
...
+    public UserDto updateUser(
+            @PathVariable(value = "id") Long userId,
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/controllers/UserController.java:45]: let's remove it

File: src/main/java/com/codesarge/banking/controllers/UserController.java [hunk 3]
```diff
@@ -28,9 +28,32 @@ public UserController(UserService userService) {
...
+            @RequestBody User user)  {
+
+        log.info("Updating user " + id + " requested");
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/controllers/UserController.java:48]: id -> user.getId()

File: src/main/java/com/codesarge/banking/service/UserService.java [hunk 4]
```diff
@@ -5,9 +5,11 @@
 import com.codesarge.banking.repositories.UserRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/service/UserService.java:8]: unused import

File: src/main/java/com/codesarge/banking/controllers/UserController.java [hunk 5]
```diff
@@ -28,9 +28,32 @@ public UserController(UserService userService) {
...
     public UserDto addUser(@RequestBody User user) {
         log.info("Requested creating user: " + user);
         return userService.addUser(user);
     }
+
+    @PutMapping("/users/{id}")
+    public UserDto updateUser(
+            @PathVariable(value = "id") Long userId,
+            @RequestBody User user)  {
+
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/controllers/UserController.java:47]: add check for user.getId is exists

File: src/main/java/com/codesarge/banking/controllers/UserController.java [hunk 6]
```diff
@@ -28,9 +28,32 @@ public UserController(UserService userService) {
...
+        log.info("Updating user " + id + " requested");
+        return userService.updateUser(user);
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/controllers/UserController.java:49]: call service.saveUser here

File: src/main/java/com/codesarge/banking/service/UserService.java [hunk 7]
```diff
@@ -31,8 +33,23 @@ public UserService(UserRepository userRepository) {
...
+    public Optional<User> findUserById(Long id) {
+         return userRepository.findById(id);
+    }
+
     public UserDto addUser(User user) {
         User saved = userRepository.save(user);
         return new UserDto(saved);
     }
+
+    public UserDto updateUser(User user) {
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/service/UserService.java:45]: remove this method

File: src/main/java/com/codesarge/banking/service/UserService.java [hunk 8]
```diff
@@ -31,8 +33,23 @@ public UserService(UserRepository userRepository) {
         return userDtos;
     }
 
+    public Optional<User> findUserById(Long id) {
+         return userRepository.findById(id);
+    }
+
     public UserDto addUser(User user) {
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/service/UserService.java:40]: rename it to saveUser

**Review by alex-anakin — CHANGES_REQUESTED** [review]

File: src/main/java/com/codesarge/banking/controllers/UserController.java [hunk 9]
```diff
@@ -5,17 +5,17 @@
 import com.codesarge.banking.service.UserService;
 import lombok.extern.log4j.Log4j2;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.ResponseEntity;
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/UserController.java:8]: done

File: src/main/java/com/codesarge/banking/controllers/UserController.java [hunk 10]
```diff
@@ -5,17 +5,17 @@
...
+import org.springframework.web.bind.annotation.*;
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/UserController.java:9]: done

File: src/main/java/com/codesarge/banking/controllers/UserController.java [hunk 11]
```diff
@@ -5,17 +5,17 @@
...
 
 import java.util.List;
+import java.util.Optional;
 
 @RestController
 @Log4j2
 public class UserController {
     private final UserService userService;
+    private Long id;
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/UserController.java:18]: deleted

File: src/main/java/com/codesarge/banking/controllers/UserController.java (hunk 1 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/UserController.java:43]: done

File: src/main/java/com/codesarge/banking/controllers/UserController.java (hunk 2 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/UserController.java:45]: done

File: src/main/java/com/codesarge/banking/service/UserService.java (hunk 4 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/service/UserService.java:8]: optimized

File: src/main/java/com/codesarge/banking/controllers/UserController.java (hunk 5 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/UserController.java:47]: fixed method

File: src/main/java/com/codesarge/banking/controllers/UserController.java (hunk 6 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/UserController.java:49]: done

File: src/main/java/com/codesarge/banking/service/UserService.java (hunk 7 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/service/UserService.java:45]: done

File: src/main/java/com/codesarge/banking/service/UserService.java (hunk 8 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/service/UserService.java:40]: done

File: src/main/java/com/codesarge/banking/controllers/UserController.java (hunk 3 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/UserController.java]: done

**Review by alex-anakin — APPROVED** [review]

**Pull request merged**